src/vsr/grid.zig

```diff
@@ -957,6 +957,10 @@ pub fn GridType(comptime Storage: type) type {
             if (!header.valid_checksum_body(block_body)) return .invalid_checksum_body;
             if (header.checksum != expect.checksum) return .unexpected_checksum;
 
+            if (constants.verify) {
+                assert(stdx.zeroed(block[header.size..vsr.sector_ceil(header.size)]));
```

Comment:
Why do we go only to the sector_ceil, rather than to the end of the block? I think we always read/write whole blocks, so we should care about padding after the sector boundary as well?

Reply:
:thinking: Is the current behavior (write full block even when `block.header.size < constants.block_size`) desirable? I don't really understand the purpose. Is it worth the write amplification for small blocks?

Reply:
Writing / reading full blocks for sure seems simpler, but, yeah, this doesn't feel optimal. 

OTOH, optimizing for blocks always being full also makes sense, as, if writing partial blocks is a big win, the problem is that we have too much partial blocks.